### PR TITLE
feat(checkbox): EC revamp for V4 - FRONT-4069

### PR DESCRIPTION
**Pull request by planctus**

### CODE REVIEW

File: src/implementations/twig/components/checkbox/checkbox-item.html.twig [hunk 1]
```diff
@@ -36,6 +37,7 @@
 {% set _helper_text = helper_text|default('') %}
 {% set _label = label|default('') %}
 {% set _css_class = 'ecl-checkbox' %}
+{% set _required_text = required_text|default('') %}
```
Comment (emeryro) [inline src/implementations/twig/components/checkbox/checkbox-item.html.twig:40]: Shouldn't it be '*' by default?
Also we could document this new parameter in the migration guide

File: src/implementations/vanilla/components/checkbox/_checkbox.scss [hunk 2]
```diff
@@ -130,52 +118,70 @@ $_label-color: null !default;
...
-      outline: 2px solid map.get(theme.$color, 'blue-100');
-      outline-offset: 2px;
+      @if map-has-key($checkbox, 'hover-circle') {
+        &::before {
+          @each $property, $value in map.get($checkbox, 'hover-circle') {
+            #{$property}: $value;
+          }
+        }
+      } @else {
+        outline: 2px solid map.get($theme, 'color' 'blue-100');
```
Comment (emeryro) [inline src/implementations/vanilla/components/checkbox/_checkbox.scss]: We're not supposed to use color blue-100 anymore (or any non semantic colors). I assume that it should be --ecl-color-primary

File: src/implementations/vanilla/components/checkbox/_checkbox.scss [hunk 3]
```diff
@@ -130,52 +118,70 @@ $_label-color: null !default;
...
     }
   }
 
   &:checked {
     ~ .ecl-checkbox__label {
       .ecl-checkbox__box:not(.ecl-checkbox__box--disabled) {
-        outline: 2px solid map.get(theme.$color, 'blue-100');
-        outline-offset: 2px;
+        @if not map-has-key($checkbox, 'hover-circle') {
+          outline: 2px solid map.get($theme, 'color' 'blue-100');
```
Comment (emeryro) [inline src/implementations/vanilla/components/checkbox/_checkbox.scss]: Same here

**Review by emeryro — CHANGES_REQUESTED** [review]

File: src/implementations/twig/components/checkbox/checkbox-item.html.twig (hunk 1 above)
Comment (planctus) [reply src/implementations/twig/components/checkbox/checkbox-item.html.twig:40]: Changed a bit the logic, we never set a default for the required text so also in this case we define it as an empty string and we expect a `required_text` prop to be defined in the checkbox object when there is a single, required, checkbox.
An item has been added in the migration doc.

**Review by emeryro — APPROVED** [review]

**Pull request merged**